Realm.Shared/linq/RealmResults.cs

```diff
@@ -124,12 +124,15 @@ private ResultsHandle CreateResultsHandle()
             {
                 return _realm.MakeResultsForTable(_targetMetadata);
             }
-            // do all the LINQ expression evaluation to build a query
-            var qv = _provider.MakeVisitor();
-            qv.Visit(Expression);
-            var queryHandle = qv._coreQueryHandle; // grab out the built query definition
-            var sortHandle = qv._optionalSortOrderHandle;
-            return _realm.MakeResultsForQuery(ObjectSchema, queryHandle, sortHandle);
+            else
```

Comment:
This  doesn't need to be an else block unless for some reason you prefer them aesthetically. When I see a return and then an else I wonder if there was a mistake and look twice at the code.